bundles/org.openhab.binding.luxtronikheatpump/src/main/java/org/openhab/binding/luxtronikheatpump/internal/LuxtronikHeatpumpHandlerFactory.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.luxtronikheatpump.internal;
+
+import static org.openhab.binding.luxtronikheatpump.internal.LuxtronikHeatpumpBindingConstants.*;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseThingHandlerFactory;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerFactory;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * The {@link LuxtronikHeatpumpHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Stefan Giehl - Initial contribution
+ */
+@NonNullByDefault
+@Component(configurationPid = "binding.luxtronikheatpump", service = ThingHandlerFactory.class)
+public class LuxtronikHeatpumpHandlerFactory extends BaseThingHandlerFactory {
+
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Set.of(THING_TYPE_HEATPUMP);
+    private static final Map<String, LuxtronikHeatpumpHandler> THING_HANDLERS = new HashMap<>();
```

Comment:
No. It actually maps the things to the handler instances. That needs to be static

Reply:
Sorry, this is wrong knowing a bit the framework and how static variables are initialised and destroyed. `getHandler(String)` is not used by the framework so it is not required there. Static variables do not have a defined lifetime. This is bad as it could lead to side effects and cause other trouble (de facto you keep instances beside of the framework alive which is very bad). Please use the frameworks possibilities for these purposes.

You are using your `getHandler(String)` only in `ChannelUpdaterJob`. There you don't even need to access the `LuxtronikHeatpumpHandlerFactory` as you already have access to `thing`. `Thing` (which `thing` is an instance of) has a non-static method `getHandler()` which just links to the proper instance. Using it this way (and properly removing the schedule when disposing the Handler) keeps the Lifecycle working as defined by openHAB.

Reply:
changed that